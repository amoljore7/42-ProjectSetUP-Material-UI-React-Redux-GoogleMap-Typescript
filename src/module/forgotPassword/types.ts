export const LOADING = 'LOADING';
export const FORGOT_PASSWORD = 'FORGOT_PASSWORD';
export const FORGOT_PASSWORD_SUCCESS = 'FORGOT_PASSWORD_SUCCESS';
export const FORGOT_PASSWORD_FAIL = 'FORGOT_PASSWORD_FAIL';

export interface Loading {
  type: typeof LOADING;
}

export interface Forgot_Password {
  type: typeof FORGOT_PASSWORD;
}

export interface LoginSuccess {
  type: typeof FORGOT_PASSWORD_SUCCESS;
  payload: SuccessPayloadType;
}

export type SuccessPayloadType = {
  email: string;
};

export interface LoginFail {
  type: typeof FORGOT_PASSWORD_FAIL;
  payload: string;
}

export type LoginDispatch = Loading | Forgot_Password | LoginSuccess | LoginFail;
