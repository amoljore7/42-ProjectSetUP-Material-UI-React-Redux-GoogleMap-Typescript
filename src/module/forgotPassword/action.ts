import { FORGOT_PASSWORD } from './types';

export interface userData {
  email: string;
}
export interface userPasswordReturnType {
  type: string;
  payload: userData;
}

export const forgotPassword: (user: userData) => userPasswordReturnType = (user: userData) => {
  return {
    type: FORGOT_PASSWORD,
    payload: user
  };
};
