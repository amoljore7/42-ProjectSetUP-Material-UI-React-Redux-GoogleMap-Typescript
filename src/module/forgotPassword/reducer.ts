import { LOADING, FORGOT_PASSWORD_SUCCESS, FORGOT_PASSWORD_FAIL } from './types';

export interface DefaultState {
  loading: boolean;
  error?: string;
}

const defaultState: DefaultState = {
  loading: false,
  error: ''
};

interface actionType {
  type: string;
  payload: string;
}

export const forgotPassword = (
  state: DefaultState = defaultState,
  action: actionType
): DefaultState => {
  switch (action.type) {
    case LOADING:
      return {
        ...state,
        loading: true
      };
    case FORGOT_PASSWORD_SUCCESS:
      return {
        ...state,
        loading: false,
        error: ''
      };
    case FORGOT_PASSWORD_FAIL:
      return {
        ...state,
        loading: false,
        error: action.payload
      };

    default:
      return state;
  }
};
