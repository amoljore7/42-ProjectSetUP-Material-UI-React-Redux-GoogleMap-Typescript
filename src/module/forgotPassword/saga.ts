import {
  call,
  CallEffectDescriptor,
  ForkEffect,
  put,
  PutEffect,
  SimpleEffect,
  takeEvery
} from 'redux-saga/effects';
import Axios, { AxiosResponse } from 'axios';
import { AppConfig } from '../../config/app-config';
import { FORGOT_PASSWORD, LOADING, FORGOT_PASSWORD_SUCCESS, FORGOT_PASSWORD_FAIL } from './types';

interface payloadData {
  type: string;
  payload: {
    email: string;
  };
}

interface forgotUserPasswordResponseType {
  id: string;
  message: string;
  status: string;
  statusCode: string;
}

export function* forgotUserPassword(
  payloadData: payloadData
): Generator<
  | SimpleEffect<'CALL', CallEffectDescriptor<unknown>>
  | PutEffect<{
      type: string;
    }>
  | Promise<forgotUserPasswordResponseType>
> {
  yield put({ type: LOADING });

  try {
    const response = (yield call(
      Axios.post,
      `${AppConfig.serverUrl}/users/forgotPassword`,
      payloadData.payload
    )) as AxiosResponse<forgotUserPasswordResponseType>;

    yield put({ type: FORGOT_PASSWORD_SUCCESS });

    if (response) {
      window.location.href = '/';
    }
  } catch (error) {
    yield put({ type: FORGOT_PASSWORD_FAIL, payload: error.message });
  }
}

export function* watchForgotPassword(): Generator<ForkEffect<never> | string> {
  yield takeEvery(FORGOT_PASSWORD, forgotUserPassword);
}
