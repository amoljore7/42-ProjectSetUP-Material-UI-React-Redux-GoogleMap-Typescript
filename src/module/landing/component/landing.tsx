import React from 'react';
import { RouteComponentProps } from 'react-router-dom';
import { Container, Button, CssBaseline, makeStyles, Typography } from '@material-ui/core';
import { MuiThemeProvider, createMuiTheme } from '@material-ui/core/styles';

const useStyles = makeStyles((theme) => ({
  headingStyle: {
    font: 'normal normal bold 60px/80px Futura',
    letterSpacing: '0px',
    color: '#000000',
    opacity: 1,
    textAlign: 'center'
  },
  headings: {
    font: 'normal normal medium 20px/27px Futura',
    color: '#707070',
    textAlign: 'center'
  },
  headings2: {
    color: '#707070',
    font: 'normal normal bold 30px/40px Futura',
    marginTop: '40px',
    marginBottom: '10px',
    textAlign: 'center'
  },
  wrapperStyle: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    marginTop: theme.spacing(10)
  },
  link: {
    fontSize: '20px'
  },
  signIn: {
    margin: theme.spacing(3, 0, 2)
  }
}));
const themeDark = createMuiTheme({
  palette: {
    background: {
      default: '#F5F5FA'
    }
  }
});

type Props = RouteComponentProps;

const LandingPage: React.FC<Props> = (props: Props) => {
  const clickHandler = () => {
    props.history.push('/login');
  };

  const classes = useStyles();
  return (
    <MuiThemeProvider theme={themeDark}>
      <Container component="main" maxWidth="xs">
        <CssBaseline />
        <div data-testid="landingWrapper" className={classes.wrapperStyle}>
          <Typography component="h1" variant="h5" className={classes.headingStyle}>
            Healthful
          </Typography>
          <Typography className={classes.headings2}>You are accessing PHI</Typography>
          <h5 className={classes.headings}>
            By clicking ’Sign in’ you agree that you will only use protected health information
            (PHI) to provide, coordinate or manage healthcare treatment for this patient
          </h5>
          <Button
            className={classes.signIn}
            style={{ backgroundColor: '#4491E0', color: '#FFFFFF' }}
            variant="contained"
            onClick={clickHandler}>
            Sign In
          </Button>
        </div>
      </Container>
    </MuiThemeProvider>
  );
};

export default LandingPage;
