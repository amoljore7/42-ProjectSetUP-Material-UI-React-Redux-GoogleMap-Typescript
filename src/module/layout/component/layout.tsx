import React from 'react';
import { Route, RouteComponentProps, Switch, Redirect } from 'react-router-dom';
import { connect } from 'react-redux';
import Navbar from '../../../components/navbar';
import rootReducer from '../../../root-reducer';

import ResourceRouting from '../../resources/routing';
import PageNotFound from '../../pageNotFound/pageNotFound';
const Home = () => <h1>Home</h1>;
const Dashboard = () => <h1>Dashboard</h1>;
const Patients = () => <h1>Patients</h1>;
const Analytics = () => <h1>Analytics</h1>;

type Props = RouteComponentProps;
const Layout: React.FC<Props> = (props: Props) => {
  return (
    <React.Fragment>
      <Navbar />
      <Switch>
        <Route
          exact
          strict
          path={props.match.url}
          render={() => <Redirect to={`${props.match.url}/resources`} />}
        />
        <Route path={`${props.match.url}/resources`} component={ResourceRouting} />
        <Route path={`${props.match.url}/home`} component={Home} />
        <Route path={`${props.match.url}/dashboard`} component={Dashboard} />
        <Route path={`${props.match.url}/patients`} component={Patients} />
        <Route path={`${props.match.url}/analytics`} component={Analytics} />
        <Route component={PageNotFound} />
      </Switch>
    </React.Fragment>
  );
};

const mapStateToProps = (state: ReturnType<typeof rootReducer>) => ({
  loginReducer: state.loginReducer
});

export default connect(mapStateToProps, {})(Layout);
