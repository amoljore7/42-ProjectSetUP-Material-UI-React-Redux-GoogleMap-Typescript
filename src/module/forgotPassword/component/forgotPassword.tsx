import React, { useState, useEffect, FormEvent } from 'react';
import { connect } from 'react-redux';
import { RouteComponentProps, withRouter } from 'react-router-dom';
import {
  MuiThemeProvider,
  createMuiTheme,
  Container,
  Button,
  CssBaseline,
  TextField,
  Card,
  CardContent,
  Typography,
  makeStyles,
  CircularProgress,
  Box
} from '@material-ui/core';
import { forgotPassword, userData, userPasswordReturnType } from '../action';
import { DefaultState } from '../../forgotPassword/reducer';
import rootReducer from '../../../root-reducer';
import ErrorString from '../../../utils/error-string';

const useStyles = makeStyles((theme) => ({
  root: {
    '& .MuiOutlinedInput-root .MuiOutlinedInput-notchedOutline': {
      borderColor: '#4491E0'
    }
  },
  card: {
    marginTop: theme.spacing(8),
    minWidth: 275,
    padding: '20px'
  },
  title: {
    font: 'normal normal bold 24px/24px Futura',
    marginBottom: '20px',
    color: '#4491E0'
  },
  form: {
    width: '100%', // Fix IE 11 issue.
    marginTop: theme.spacing(1)
  },
  signIn: {
    color: '#707070',
    font: 'normal normal bold 26px/35px Futura'
  },
  submitBtn: {
    margin: theme.spacing(3, 0, 2),
    font: 'normal normal medium 14px/16px Roboto'
  },
  circularProgress: {
    marginLeft: 0,
    color: 'white',
    marginRight: '20px'
  },
  headings: {
    font: 'normal normal medium 20px/27px Futura',
    color: '#707070'
  },
  errorMessage: {
    color: 'red'
  }
}));
const themeDark = createMuiTheme({
  palette: {
    background: {
      default: '#F5F5FA'
    }
  }
});

interface StoreProps {
  forgotPasswordReducer: DefaultState;
}

interface DispatchProps {
  forgotPassword: (user: userData) => userPasswordReturnType;
}

type Props = StoreProps & DispatchProps & RouteComponentProps;

export const ForgotPassword: React.FC<Props> = (props: Props) => {
  const classes = useStyles();
  const [submitbtnDisabled, setSubmitBtnDisabled] = useState(false);
  const [formData, setFormData] = useState({
    email: ''
  });

  useEffect(() => {
    setSubmitBtnDisabled(true);
  }, []);

  const inputChangeHandler = (e: React.ChangeEvent<HTMLInputElement>) => {
    const { name, value } = e.target;
    setFormData((formData) => ({ ...formData, [name]: value }));
    if (!name.trim() || !value.trim()) {
      setSubmitBtnDisabled(true);
    } else {
      setSubmitBtnDisabled(false);
    }
  };
  const formSubmitHandler = (event: FormEvent<HTMLFormElement>) => {
    event.preventDefault();
    props.forgotPassword(formData);
  };

  return (
    <MuiThemeProvider theme={themeDark}>
      <Container component="main" maxWidth="xs">
        <CssBaseline />
        <Card className={classes.card} variant="outlined">
          <CardContent>
            <Typography component="h1" variant="h5" className={classes.title}>
              TeleNavigation 2.0
            </Typography>
            <Typography component="h2" variant="h5" className={classes.signIn}>
              Forgot Password
            </Typography>
            <form
              data-testid="formWrapper"
              className={classes.form}
              noValidate
              onSubmit={formSubmitHandler}>
              <h5 className={classes.headings}>
                A Verification link will be sent to your registered email address
              </h5>
              <TextField
                // className={classes.root}
                variant="outlined"
                margin="normal"
                required
                fullWidth
                type="text"
                id="email"
                label="E-mail"
                name="email"
                onChange={inputChangeHandler}
                inputProps={{
                  'data-testid': 'emailInput'
                }}
              />
              <Box>
                <Typography className={classes.errorMessage}>
                  {ErrorString(props.forgotPasswordReducer.error || '') ||
                    props.forgotPasswordReducer.error}
                </Typography>
              </Box>
              <Button
                data-testid="forgotPasswordButton"
                className={classes.submitBtn}
                type="submit"
                // fullWidth
                variant="contained"
                style={{ backgroundColor: '#4491E0', color: '#FFFFFF' }}
                disabled={submitbtnDisabled}>
                {props.forgotPasswordReducer.loading ? (
                  <CircularProgress
                    data-testid="forgotPasswordButtonLoader"
                    className={classes.circularProgress}
                    size={30}
                  />
                ) : (
                  'SEND VERIFICATION EMAIL'
                )}
              </Button>
            </form>
          </CardContent>
        </Card>
      </Container>
    </MuiThemeProvider>
  );
};

const mapStateToProps = (state: ReturnType<typeof rootReducer>) => ({
  forgotPasswordReducer: state.forgotPasswordReducer
});

export default connect(mapStateToProps, { forgotPassword })(withRouter(ForgotPassword));
