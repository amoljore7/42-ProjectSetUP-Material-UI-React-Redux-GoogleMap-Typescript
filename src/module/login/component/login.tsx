import React, { useState, useEffect, FormEvent } from 'react';
import { connect } from 'react-redux';
import { withRouter, RouteComponentProps } from 'react-router-dom';
import {
  Container,
  Button,
  CssBaseline,
  TextField,
  Link,
  Grid,
  Card,
  CardContent,
  Typography,
  makeStyles,
  CircularProgress,
  InputAdornment,
  IconButton,
  Box
} from '@material-ui/core';
import { Visibility, VisibilityOff } from '@material-ui/icons';
import { MuiThemeProvider, createMuiTheme } from '@material-ui/core/styles';

import ErrorString from '../../../utils/error-string';

import { userLoginReturnType, userData } from '../action';
import { DefaultState } from '../reducer';
import rootReducer from '../../../root-reducer';
import { userLogin } from '../action';
import { getCurrentUser } from '../../../utils/local-storage';

const useStyles = makeStyles((theme) => ({
  root: {
    '& .MuiOutlinedInput-root .MuiOutlinedInput-notchedOutline': {
      borderColor: '#4491E0'
    }
  },
  card: {
    marginTop: theme.spacing(8),
    minWidth: 275,
    padding: '20px'
  },
  title: {
    font: 'normal normal bold 24px/24px Futura',
    marginBottom: '20px',
    color: '#4491E0'
  },
  form: {
    width: '100%', // Fix IE 11 issue.
    marginTop: theme.spacing(1)
  },
  signIn: {
    color: '#707070',
    font: 'normal normal bold 26px/35px Futura'
  },
  submitBtn: {
    margin: theme.spacing(3, 0, 2),
    font: 'normal normal medium 14px/16px Roboto'
  },
  circularProgress: {
    marginLeft: 0,
    color: 'white',
    marginRight: '20px'
  },
  forgotPassword: {
    color: '#4491E0'
  },
  errorMessage: {
    color: 'red'
  }
}));
const themeDark = createMuiTheme({
  palette: {
    background: {
      default: '#F5F5FA'
    }
  }
});

interface storeProps {
  loginReducer: DefaultState;
}

interface DispatchProps {
  userLogin: (user: userData) => userLoginReturnType;
}

type Props = storeProps & RouteComponentProps & DispatchProps;

export const LoginPage: React.FC<Props> = (props: Props) => {
  const classes = useStyles();
  const [submitbtnDisabled, setSubmitBtnDisabled] = useState(false);
  const [showPassword, setShowPassword] = useState(false);
  const [formData, setFormData] = useState({
    username: '',
    password: ''
  });
  const clickHandlerEmail = () => {
    props.history.push('/forgot-password');
  };

  useEffect(() => {
    if (getCurrentUser()) {
      props.history.push('user');
    }
  }, [props.loginReducer]);
  useEffect(() => {
    setSubmitBtnDisabled(true);
  }, []);
  const handleClickShowPassword = () => {
    setShowPassword(!showPassword);
  };

  const inputChangeHandler = (e: React.ChangeEvent<HTMLInputElement>) => {
    const { name, value } = e.target;
    setFormData((formData) => ({ ...formData, [name]: value }));
    if (!formData.username.trim() || !formData.password.trim() || !value.trim()) {
      setSubmitBtnDisabled(true);
    } else {
      setSubmitBtnDisabled(false);
    }
  };
  const formSubmitHandler = (event: FormEvent<HTMLFormElement>) => {
    event.preventDefault();
    props.userLogin(formData);
  };

  return (
    <MuiThemeProvider theme={themeDark}>
      <Container component="main" maxWidth="xs">
        <CssBaseline />
        <Card className={classes.card} variant="outlined">
          <CardContent>
            <Typography component="h1" variant="h5" className={classes.title}>
              TeleNavigation 2.0
            </Typography>
            <Typography component="h2" variant="h5" className={classes.signIn}>
              Sign In
            </Typography>
            <form
              data-testid="formWrapper"
              className={classes.form}
              noValidate
              onSubmit={formSubmitHandler}>
              <TextField
                variant="outlined"
                margin="normal"
                required
                fullWidth
                id="username"
                label="Username"
                name="username"
                autoComplete="username"
                onChange={inputChangeHandler}
                inputProps={{
                  'data-testid': 'username'
                }}
              />
              <TextField
                variant="outlined"
                margin="normal"
                required
                fullWidth
                name="password"
                label="Password"
                type={showPassword ? 'text' : 'password'}
                id="password"
                autoComplete="current-password"
                onChange={inputChangeHandler}
                inputProps={{
                  'data-testid': 'password'
                }}
                InputProps={{
                  endAdornment: (
                    <InputAdornment position="end">
                      <IconButton
                        edge="end"
                        aria-label="toggle password visibility"
                        onClick={handleClickShowPassword}>
                        {showPassword ? <VisibilityOff /> : <Visibility />}
                      </IconButton>
                    </InputAdornment>
                  )
                }}
              />
              <Box>
                <Typography className={classes.errorMessage}>
                  {ErrorString(props.loginReducer.error) || props.loginReducer.error}
                </Typography>
              </Box>
              <Button
                data-testid="loginButton"
                className={classes.submitBtn}
                type="submit"
                // fullWidth
                variant="contained"
                style={{ backgroundColor: '#4491E0', color: '#FFFFFF' }}
                disabled={submitbtnDisabled}>
                {props.loginReducer.loading ? (
                  <CircularProgress
                    data-testid="loginLoader"
                    className={classes.circularProgress}
                    size={30}
                  />
                ) : (
                  'LOGIN'
                )}
              </Button>
              <Grid container>
                <Grid item xs>
                  <Link
                    onClick={clickHandlerEmail}
                    variant="body2"
                    className={classes.forgotPassword}>
                    Forgot password?
                  </Link>
                </Grid>
              </Grid>
            </form>
          </CardContent>
        </Card>
      </Container>
    </MuiThemeProvider>
  );
};

const mapStateToProps = (state: ReturnType<typeof rootReducer>) => ({
  loginReducer: state.loginReducer
});

export default connect(mapStateToProps, { userLogin })(withRouter(LoginPage));
