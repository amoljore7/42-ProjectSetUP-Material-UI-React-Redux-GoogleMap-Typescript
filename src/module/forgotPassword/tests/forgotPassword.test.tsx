import React from 'react';
import '@testing-library/jest-dom';
import { render, fireEvent, waitFor, screen } from '@testing-library/react';

import { ForgotPassword } from '../component/forgotPassword';

describe('forgot password page', () => {
  it('loads and displays forgot password page', async () => {
    const mockProps = {
      history: {} as any,
      location: {} as any,
      match: {} as any,
      forgotPasswordReducer: {} as any,
      forgotPassword: jest.fn()
    };
    const { asFragment } = render(<ForgotPassword {...mockProps} />);
    expect(asFragment()).toMatchSnapshot();
    expect(screen.getByTestId('formWrapper').children.length).toEqual(4);

    const forgotPasswordButton = screen.getByTestId('forgotPasswordButton');
    expect(forgotPasswordButton).toBeDisabled();

    const emailInput = screen.getByTestId('emailInput');

    fireEvent.change(emailInput, { target: { value: 'abc.xyz@gmail.com' } });

    expect(forgotPasswordButton).toBeEnabled();
  });
  it('loader is displayed on forgot password button when it is clicked', () => {
    const mockProps = {
      history: {} as any,
      location: {} as any,
      match: {} as any,
      forgotPasswordReducer: {
        loading: true,
        error: ''
      },
      forgotPassword: jest.fn()
    };
    render(<ForgotPassword {...mockProps} />);
    expect(screen.getByTestId('forgotPasswordButtonLoader')).toBeInTheDocument();
  });
  it('loader is not displayed on forgot password button when it is not clicked', () => {
    const mockProps = {
      history: {} as any,
      location: {} as any,
      match: {} as any,
      forgotPasswordReducer: {
        loading: false,
        error: ''
      },
      forgotPassword: jest.fn()
    };
    render(<ForgotPassword {...mockProps} />);
    expect(screen.getByText('SEND VERIFICATION EMAIL')).toBeInTheDocument();
  });
});
