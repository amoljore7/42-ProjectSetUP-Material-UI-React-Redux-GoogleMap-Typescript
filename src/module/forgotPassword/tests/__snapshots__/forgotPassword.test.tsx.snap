// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`forgot password page loads and displays forgot password page 1`] = `
<DocumentFragment>
  <main
    class="MuiContainer-root MuiContainer-maxWidthXs"
  >
    <div
      class="MuiPaper-root MuiCard-root makeStyles-card-2 MuiPaper-outlined MuiPaper-rounded"
    >
      <div
        class="MuiCardContent-root"
      >
        <h1
          class="MuiTypography-root makeStyles-title-3 MuiTypography-h5"
        >
          TeleNavigation 2.0
        </h1>
        <h2
          class="MuiTypography-root makeStyles-signIn-5 MuiTypography-h5"
        >
          Forgot Password
        </h2>
        <form
          class="makeStyles-form-4"
          data-testid="formWrapper"
          novalidate=""
        >
          <h5
            class="makeStyles-headings-8"
          >
            <address>
              A Verification link will be sent to your registered email address
            </address>
             
          </h5>
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
              data-shrink="false"
              for="email"
              id="email-label"
            >
              E-mail
              <span
                aria-hidden="true"
                class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 *
              </span>
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input"
                data-testid="emailInput"
                id="email"
                name="email"
                required=""
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-10 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-12"
                >
                  <span>
                    E-mail *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiBox-root MuiBox-root-14"
          >
            <p
              class="MuiTypography-root makeStyles-errorMessage-9 MuiTypography-body1"
            />
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submitBtn-6 Mui-disabled MuiButton-fullWidth Mui-disabled"
            data-testid="forgotPasswordButton"
            disabled=""
            style="background-color: rgb(68, 145, 224);"
            tabindex="-1"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              SEND VERIFICATION EMAIL
            </span>
          </button>
        </form>
      </div>
    </div>
  </main>
</DocumentFragment>
`;
